gp_learner.py

```diff
@@ -684,6 +686,121 @@ def mutate_fix_var(
     ]
     return res
 
+def mutate_deep_narrow(
+        sparql,
+        timeout,
+        child,
+        gtp_scores,
+        dn_path_steps_max_n=config.MUTPB_DN_PS_MAX_N,
+        direct=None, 
+        childin=False,
+        limit=config.MUTPB_FV_QUERY_LIMIT,  # TODO: Limit benutzen?
+):
+    if not child.matching_node_pairs:
+        ev = evaluate(
+            sparql, timeout, gtp_scores, child)  # TODO: Muss hier run/gen dazu?
+        update_individuals([child], [ev])
+    gtps = child.matching_node_pairs
+    if not gtps:
+        return [child]
+    #TODO: testen, wie die Verteilung gut ist
+    n = random.choice(range(dn_path_steps_max_n))+1
+    n = 2
+    node = [SOURCE_VAR]
```

Comment:
why not simply `nodes = [SOURCE_VAR] + [Variable('n%d' % i) for i in range(n)] + [TARGET_VAR]`?